lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -1785,11 +1785,28 @@ DEFINE_ISXLOCALEAVAILABLE(PR, uloc)
         return nullptr;
 #else
         EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
-        INTL_CHECK_ARGS(args.Info.Count == 4 && JavascriptString::Is(args[1]) && JavascriptString::Is(args[2]) && DynamicObject::Is(args[3]));
+        INTL_CHECK_ARGS(
```

Comment:
This is a failfast. We went with this route because these functions are only called from Intl.js, which should be massaging user-provided inputs such that these assumptions are valid. If not, its a bug in Intl.js code.